wandb/sdk/wandb_run.py

```diff
@@ -925,6 +931,11 @@ def log(
             raise ValueError("Key values passed to `wandb.log` must be strings.")
 
         if step is not None:
+            if len(wandb.patched["tensorboard"]) > 0:
```

Comment:
```
using_tensorboard = len(wandb.patched["tensorboard"]) == 0
if using_tensorboard:
```
Im not sure what is happening here.. you are warning that step cant bee set, but then continuing on?
Is that because we are stripping step later?
seems we might end up hitting other termwarns  lateer

Reply:
Yeah, we warn once here that you can't use step with tensorboard, I haven't seen other term warns later in my testing, did you have any in mind?

Reply:
The warn right below this one.
What would it mean if that fires.
This would happen if the user logged:
```
wandb.log(metrics, step=-10000)
```
What would you expect would happen in this case.
When we had talked about it I was under the impression that the plan was to
warn and ignore the user supplied step if using tensorboard_sync

This code suggests otherwise, we do a:
```
            if self.history._step > step:
```
and then drop the entire wandb.log() in this case, Is that the desired result?

And right below that we do:
```
            elif step > self.history._step:
                self.history._flush()
                self.history._step = step
```

So in this case if the user passes step like:
```
wandb.log(metrics, step=100000000000)
```

This code will:
- warn them step cannot be set in the case of tensorboard
- But still use the step they passed to close out the previous history if step was set
- record the step in the history
- And i think the code will later remove the step that was just added to the history

*I think the code is correct*, and i am not suggesting changing the logic, but a reader of the code would be very confused.

Maybe a comment that says:
```
if step is passed in when tensorboard_sync is used we
honor the step passed to make decisions about how to close out the history record, but will strip this history later on in publish_history()
```

And then in handle history is the other side of the path, a comment in assign_step telling what we are doing there.   What do we expect in this path (step isnt in history and we are assigning it here, but we honor when step was assigned in history for old logs)